### PR TITLE
Support multi-comp fin set shape editing

### PR DESCRIPTION
**Pull request by SiboVG**
This PR support multi-component fin set shape editing.

The basic principle is:
1. Editing the first fin point of the main fin set will edit the first fin point of the listener fin set
2. Editing the last fin point of the main fin set will edit the last fin point of the listener fin set
3. Editing intermediate fin points of the main fin set will edit the correspondingly indexed fin points of the listener fin set
    1. If the main fin set has more points than the listener fin set, then ignore the out-of-bound points for the listener fin set
    2. If the main fin set has more points than the listener fin set, e.g. main set has 5 points, listener set 4 points, if you edit point 4 of the main fin set (second-to-last point if the main set, last point of the listener set), then point 4 of the listener fin set will **not** be edited. You can only edit the last point of the listener set by editing the last point of the main fin set.

Basically a lot of rules that try to say: "Just use multi-comp fin set shape editing for fins that have the same shape".

https://user-images.githubusercontent.com/11031519/205123751-037f18f7-aa4f-4dbb-9cdb-cb41008396bd.mp4

### CODE REVIEW

**Comment by SiboVG** [comment]
@neilweinstock made the remark that this behavior may be too complex for users to phantom and we should protect them from messing things up by hiding the "Shape" panel when multi-editing freeform fin sets. I'll leave that here as a discussion topic.

**Comment by hcraigmiller** [comment]
Functions as described; no anomalies found as to the fin-point change capabilities described. I presume that this multi-component editing PR is intended only to change the fin shape, not to increase the fin count, split fins, or change fin types (such as changing trapezoidal to freeform) this PR does none of these.

I disagree with @neilweinstock regarding hiding the "Shape" panel when multi-editing freeform fin sets. First, I'm not sure that an objective in developing OpenRocket should be to dumb-down its capabilities to "protect" users from "messing things up." I have always believed, and still do believe that an important element of OpenRocket is to educate, to improve a user's knowledge and abilities, not intentionally suppress that opportunity.

Second, this feature would have tremendous benefits to those who regularly use freeform fin sets. Again, I see no need to blunt the capabilities of OpenRocket for one user because another may not understand how to correctly use it. Look at the overrides, talk about an area that is rarely fully understood and has the potential for huge mistakes by less knowledgeable users.

What's the saying, we learn from our mistakes. My thoughts.

Build 1247
[Windows 11 Pro; Version 22H2; OS Build 22621.521; Windows Feature Experience Pack 1000.22634.1000.0]
[Java "11.0.15" 2022-04-19 LTS; Java(TM) SE Runtime Environment 18.9 (build 11.0.15+8-LTS-149)]

**Comment by neilweinstock** [comment]
> I disagree with @neilweinstock regarding hiding the "Shape" panel when multi-editing freeform fin sets. First, I'm not sure that an objective in developing OpenRocket should be to dumb-down its capabilities to "protect" users from "messing things up." I have always believed, and still do believe that an important element of OpenRocket is to educate, to improve a user's knowledge and abilities, not intentionally suppress that opportunity.
> 
> Second, this feature would have tremendous benefits for those who regularly use freeform fin sets. Again, I see no need to blunt a capability of OpenRocket for one user because another may not understand how to correctly use it. Look at the overrides, talk about an area that is rarely fully understood and has the potential for huge (possibly catastrophic) mistakes by less knowledgeable users.

I'm not convinced it's so useful, especially when the fins are not the exact same shape to begin with.

Also, in general, I would not equate "dumbing down" with "removing features that are pointlessly dangerous and/or confusing, with low benefit".  Learning how to navigate software pitfalls and/or poorly implemented features is not the sort of "improving a user's knowledge" that I consider to be worthwhile.

However, in this particular case, although the proposed functionality is weird, I don't think that it's likely to be activated by accident and cause unintended problems for a user, and therefore I'm OK with it.  And I understand the reason for the behavior, although it's more of a developer reason than a user-focused reason IMHO.

The main use I foresee for it is in managing bilateral symmetric fin sets (which of necessity must be implemented as two separate single fins, see #1548)....  In that case the different fins will always be identical.  I can't think of an instance where editing two different-shaped fins at the same time makes any sense.

**Comment by SiboVG** [comment]
I'll ask @JoePfeiffer to make the final decision on this feature.

**Comment by neilweinstock** [comment]
In case it was lost in my previous screeds, I'm OK with this feature as it currently stands.  I don't think there's any reason to hold it back.

**Comment by JoePfeiffer** [comment]
Let's put it in and see what the users say.
